### PR TITLE
Apis 1228 - Ramlise State Pension

### PR DESCRIPTION
**Pull request by antroy**

### CODE REVIEW

File: public/api/conf/1.0/schemas/state-pension.json [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            },
+            "weeklyAmount": {
+              "type": "number",
+              "description": "The forecasted amount of State Pension per week",
+              "example": "146.76"
+            },
+            "monthlyAmount": {
+              "type": "number",
+              "description": "The forecasted amount of State Pension per month",
+              "example": "638.14.1"
```
Comment (dspork) [inline public/api/conf/1.0/schemas/state-pension.json]: I think this needs to be 638.14

Comment (antroy) [inline public/api/conf/1.0/schemas/state-pension.json]: I think you are right... I'd automated it from the XML docs and missed it when I went through the schema!

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 2]
```diff
@@ -8,89 +8,89 @@
     </thead>
     <tbody>
         <tr>
-            <td><p>Not Listed</p></td>
-            <td><p></p></td>
-            <td><p>The regular response at the top of this document</p></td>
+            <td><p>Regular Taxpayer</p></td>
+            <td><p>(Any not prefixed as below)</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 3]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Exclusion</p></td>
+            <td><p>Taxpayer with the PostStatePensionAge exclusion</p></td>
             <td><p><code>PS</code></p></td>
-            <td><p>A user with the PostStatePensionAge exclusion</p></td>
+            <td><p>200 (OK) Payload as per exclusion example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as exclusion example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 4]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Exclusion</p></td>
+            <td><p>Taxpayer with the AmountDissonance exclusion</p></td>
             <td><p><code>AM</code></p></td>
-            <td><p>A user with the AmountDissonance exclusion</p></td>
+            <td><p>200 (OK) Payload as per exclusion example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as exclusion example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 5]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Exclusion</p></td>
+            <td><p>Taxpayer with the Abroad exclusion</p></td>
             <td><p><code>AB</code></p></td>
-            <td><p>A user with the Abroad exclusion</p></td>
+            <td><p>200 (OK) Payload as per exclusion example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as exclusion example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 6]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Exclusion</p></td>
+            <td><p>Taxpayer with the IsleOfMan exclusion</p></td>
             <td><p><code>MA</code></p></td>
-            <td><p>A user with the IsleOfMan exclusion</p></td>
+            <td><p>200 (OK) Payload as per exclusion example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as exclusion example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 7]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Exclusion</p></td>
+            <td><p>Taxpayer with the MarriedWomenReducedRateElection exclusion</p></td>
             <td><p><code>MW</code></p></td>
-            <td><p>A user with the MarriedWomenReducedRateElection exclusion</p></td>
+            <td><p>200 (OK) Payload as per exclusion example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as exclusion example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 8]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Exclusion</p></td>
+            <td><p>Taxpayer that is excluded for multiple reasons</p></td>
             <td><p><code>MX</code></p></td>
-            <td><p>A user that is excluded for multiple reasons</p></td>
+            <td><p>200 (OK) Payload as per exclusion example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as exclusion example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 9]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Success</p></td>
+            <td><p>Taxpayer with COPE</p></td>
             <td><p><code>CR</code></p></td>
-            <td><p>A user with COPE</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 10]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Success</p></td>
+            <td><p>Taxpayer with a protected payment</p></td>
             <td><p><code>PP</code></p></td>
-            <td><p>A user with a protected payment</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 11]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Success</p></td>
+            <td><p>Taxpayer below the minimum qualifying period who cannot get any State Pension even by working until State Pension age and filling payable gaps</p></td>
             <td><p><code>CA</code></p></td>
-            <td><p>A user below the minimum qualifying period who cannot get any State Pension even by working until State Pension age and filling payable gaps</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 12]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Success</p></td>
+            <td><p>Taxpayer below the minimum qualifying period who can get some State Pension by working until State Pension age and also filling payable gaps</p></td>
             <td><p><code>AL</code></p></td>
-            <td><p>A user below the minimum qualifying period who can get some State Pension by working until State Pension age and also filling payable gaps</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Should be: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 13]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Success</p></td>
+            <td><p>Taxpayer below the minimum qualifying period that can get some State Pension by just working to State Pension age, but can get more by filling payable gaps as well</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Taxpayer below the minimum qualifying period that can get some State Pension by working until State Pension age, but can get more by filling payable gaps as well

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 14]
```diff
@@ -8,89 +8,89 @@
...
             <td><p><code>AE</code></p></td>
-            <td><p>A user below the minimum qualifying period that can get some State Pension by just working to State Pension age, but can get more by filling payable gaps as well</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 15]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Success</p></td>
+            <td><p>Taxpayer who will not reach the full rate by working until State Pension age, but can make up the difference by filling a gap</p></td>
             <td><p><code>AH</code></p></td>
-            <td><p>A user who will not reach the full rate by working until State Pension age, but can make up the difference by filling a gap</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 16]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Success</p></td>
+            <td><p>Taxpayer who will reach the maximum by continuing to work, they will reach the maximum before their State Pension age</p></td>
             <td><p><code>AK</code></p></td>
-            <td><p>A user who will reach the maximum by continuing to work, they will reach the maximum before their State Pension age</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 17]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>Success</p></td>
+            <td><p>Taxpayer who has already reached the full State Pension rate</p></td>
             <td><p><code>AP</code></p></td>
-            <td><p>A user who has already reached the full State Pension rate</p></td>
+            <td><p>200 (OK) Payload as per regular example above</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: 200 (OK) Payload as regular example above

File: public/api/conf/1.0/sandbox/get-state-pension.md [hunk 18]
```diff
@@ -8,89 +8,89 @@
...
         </tr>
         <tr>
-            <td><p>EXCLUSION_DEAD</p></td>
+            <td><p>Taxpayer who is dead</p></td>
```
Comment (fred-jackson) [inline public/api/conf/1.0/sandbox/get-state-pension.md]: Taxpayer who has a date of death

File: public/api/conf/1.0/schemas/state-pension.json [hunk 19]
```diff
@@ -0,0 +1,180 @@
...
+  "type": "object",
+  "properties": {
+    "exclusionReasons": {
+      "type": "array",
+      "items": {
+      "type": "string",
+        "oneOf": [
+          {
+            "enum": [ "Abroad" ],
+            "description": "Male who has an overseas address (outside United Kingdom and Isle of Man) and retires before 06/10/2018. Unable to automate currently, manual intervention required."
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: Male who has an address outside the United Kingdom and Isle of Man, and retires before 6 October 2018. Unable to automate currently, manual intervention required.

File: public/api/conf/1.0/schemas/state-pension.json [hunk 20]
```diff
@@ -0,0 +1,180 @@
...
+          "type": "boolean",
+          "description": "Whether the customer has an amount higher than the full rate at the end of the 2015/16 tax year",
+          "example": "false"
+        },
+        "current": {
+          "type": "object",
+          "properties": {
+            "weeklyAmount": {
+              "type": "number",
+              "description": "The amount of accrued State Pension to date per week",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The amount of accrued State Pension to date each week

File: public/api/conf/1.0/schemas/state-pension.json [hunk 21]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "133.41"
+            },
+            "monthlyAmount": {
+              "type": "number",
+              "description": "The amount of accrued State Pension to date per month",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The amount of accrued State Pension to date each month

File: public/api/conf/1.0/schemas/state-pension.json [hunk 22]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "580.1"
+            },
+            "annualAmount": {
+              "type": "number",
+              "description": "The amount of accrued State Pension to date per year",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The amount of accrued State Pension to date each year

File: public/api/conf/1.0/schemas/state-pension.json [hunk 23]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "6961.14"
+            }
+          }
+        },
+        "forecast": {
+          "type": "object",
+          "properties": {
+            "yearsToWork": {
+              "type": "number",
+              "description": "The number of future qualifying years needed to achieve the forecast amount",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The number of future qualifying years needed to get the forecast amount

File: public/api/conf/1.0/schemas/state-pension.json [hunk 24]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "1"
+            },
+            "weeklyAmount": {
+              "type": "number",
+              "description": "The forecasted amount of State Pension per week",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The forecasted amount of State Pension each week

File: public/api/conf/1.0/schemas/state-pension.json [hunk 25]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "146.76"
+            },
+            "monthlyAmount": {
+              "type": "number",
+              "description": "The forecasted amount of State Pension per month",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The forecasted amount of State Pension each month

File: public/api/conf/1.0/schemas/state-pension.json [hunk 26]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "638.14.1"
+            },
+            "annualAmount": {
+              "type": "number",
+              "description": "The forecasted amount of State Pension per year",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The forecasted amount of State Pension each year

File: public/api/conf/1.0/schemas/state-pension.json [hunk 27]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "7657.73"
+            }
+          }
+        },
+        "maximum": {
+          "type": "object",
+          "properties": {
+            "yearsToWork": {
+              "type": "number",
+              "description": "The number of future qualifying years needed to achieve the maximum amount",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The number of future qualifying years needed to get the maximum amount

File: public/api/conf/1.0/schemas/state-pension.json [hunk 28]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "3"
+            },
+            "gapsToFill": {
+              "type": "number",
+              "description": "The minimum number of National Insurance gaps that need to be filled to achieve the maximum amount",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The minimum number of National Insurance gaps that need to be filled to get the most possible

File: public/api/conf/1.0/schemas/state-pension.json [hunk 29]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "2"
+            },
+            "weeklyAmount": {
+              "type": "number",
+              "description": "The maximum amount of State Pension per week",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The maximum amount of State Pension each week

File: public/api/conf/1.0/schemas/state-pension.json [hunk 30]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "155.65"
+            },
+            "monthlyAmount": {
+              "type": "number",
+              "description": "The maximum amount of State Pension per month",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The maximum amount of State Pension each month

File: public/api/conf/1.0/schemas/state-pension.json [hunk 31]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "676.8"
+            },
+            "annualAmount": {
+              "type": "number",
+              "description": "The maximum amount of State Pension per year",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The maximum amount of State Pension each year

File: public/api/conf/1.0/schemas/state-pension.json [hunk 32]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "8121.59"
+            }
+          }
+        },
+        "cope": {
+          "type": "object",
+          "properties": {
+            "weeklyAmount": {
+              "type": "number",
+              "description": "The amount of Contracted Out Pension Equivalent per week",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The amount of Contracted Out Pension Equivalent each week

File: public/api/conf/1.0/schemas/state-pension.json [hunk 33]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "0"
+            },
+            "monthlyAmount": {
+              "type": "number",
+              "description": "The amount of Contracted Out Pension Equivalent per month",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The amount of Contracted Out Pension Equivalent each month

File: public/api/conf/1.0/schemas/state-pension.json [hunk 34]
```diff
@@ -0,0 +1,180 @@
...
+              "example": "0"
+            },
+            "annualAmount": {
+              "type": "number",
+              "description": "The amount of Contracted Out Pension Equivalent per year",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The amount of Contracted Out Pension Equivalent each year

File: public/api/conf/1.0/schemas/state-pension.json [hunk 35]
```diff
@@ -0,0 +1,180 @@
...
+      "example": "64"
+    },
+    "pensionDate": {
+      "type": "string",
+      "description": "The date the customer can claim State Pension",
+      "example": "2018-07-06"
+    },
+    "finalRelevantYear": {
+      "type": "string",
+      "description": "The last full Tax Year before the customer\u2019s State Pension Date, for example 2017/18 with a date of 06/07/2018",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The last full tax year before the customer\u2019s State Pension date. For example, 2017/18 with a date of 6 July 2018

File: public/api/conf/1.0/schemas/state-pension.json [hunk 36]
```diff
@@ -0,0 +1,180 @@
...
+      "example": "2017-18"
+    },
+    "numberOfQualifyingYears": {
+      "type": "number",
+      "description": "The number of qualifying years of National Insurance",
+      "example": "30"
+    },
+    "pensionSharingOrder": {
+      "type": "boolean",
+      "description": "Whether the customer has a Pension Sharing Order in effect",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: Whether the customer has a 'pension sharing order' in effect

File: public/api/conf/1.0/schemas/state-pension.json [hunk 37]
```diff
@@ -0,0 +1,180 @@
...
+      "example": "true"
+    },
+    "currentWeeklyPensionAmount": {
+      "type": "number",
+      "description": "The full rate of State Pension per week",
```
Comment (fred-jackson) [inline public/api/conf/1.0/schemas/state-pension.json]: The full rate of State Pension each week

**Pull request merged**